File: 通用代码/装饰模式/ConcreteDecorator2.java
public class ConcreteDecorator2 extends Decorator{
    //定义被修饰者
    public ConcreteDecorator2(Component _component){
        super(_component);
    }
    //定义自己的修饰方法
    private void method2(){
        System.out.println("method2 修饰");
    }
    //重写父类的operate方法
    public void operate(){
        super.operate();
        this.method2();
    }
}

File: 通用代码/装饰模式(C++)/ConcreteDecorator2.cpp
#include "ConcreteDecorator2.h"

using namespace std;
ConcreteDecorator2::ConcreteDecorator2(Component* com):Decorator(com)
{}

ConcreteDecorator2::~ConcreteDecorator2()
{
    cout << "~ConcreteDecorator2" << endl;
}

void ConcreteDecorator2::Operation()
{
    this->_com->Operation();
    //附加职责B
    this->AddBehavorB();
}

void ConcreteDecorator2::AddBehavorB()
{
    cout << "附加职责B：ConcreteDecorator2::AddBehavorB" << endl;
}
